modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,671 @@
+		}
+
+		$awaiting_mod      = wp_count_comments();
+		$awaiting_mod      = $awaiting_mod->moderated;
+		$awaiting_mod_i18n = number_format_i18n( $awaiting_mod );
+		/* translators: %s: Number of comments. */
+		$awaiting_mod_text = sprintf( _n( '%s Comment in moderation', '%s Comments in moderation', $awaiting_mod, 'jetpack' ), $awaiting_mod_i18n );
+
+		/* translators: %s: Number of comments. */
+		$menu_title = sprintf( __( 'Comments %s', 'jetpack' ), '<span class="awaiting-mod count-' . absint( $awaiting_mod ) . '"><span class="pending-count" aria-hidden="true">' . $awaiting_mod_i18n . '</span><span class="comments-in-moderation-text screen-reader-text">' . $awaiting_mod_text . '</span></span>' );
+		$menu_slug  = 'https://wordpress.com/comments/all/' . $domain;
+
+		remove_menu_page( 'edit-comments.php' );
+		remove_submenu_page( 'edit-comments.php', 'edit-comments.php' );
+
+		add_menu_page( esc_attr__( 'Comments', 'jetpack' ), $menu_title, 'edit_posts', $menu_slug, null, 'dashicons-admin-comments', 25 );
+		$this->migrate_submenus( 'edit-comments.php', $menu_slug );
+	}
+
+	/**
+	 * Adds Jetpack menu.
+	 *
+	 * @param string $domain Site domain.
+	 */
+	public function add_jetpack_menu( $domain ) {
+		if ( ! $this->is_wpcom_site() && ! jetpack_is_atomic_site() ) {
+			return;
+		}
+
+		global $menu;
+
+		$position = 50;
+		while ( isset( $menu[ $position ] ) ) {
+			$position++;
+		}
+
+		// TODO: Replace with proper SVG data url.
+		$jetpack_icon = "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' viewBox='0 0 32 32' %3E%3Cpath fill='%23a0a5aa' d='M16,0C7.2,0,0,7.2,0,16s7.2,16,16,16s16-7.2,16-16S24.8,0,16,0z'%3E%3C/path%3E%3Cpolygon fill='%23fff' points='15,19 7,19 15,3 '%3E%3C/polygon%3E%3Cpolygon fill='%23fff' points='17,29 17,13 25,13 '%3E%3C/polygon%3E%3C/svg%3E";
+		$jetpack_slug = 'https://wordpress.com/activity-log/' . $domain;
+
+		$this->add_admin_menu_separator( $position++, 'manage_options' );
+		add_menu_page( esc_attr__( 'Jetpack', 'jetpack' ), __( 'Jetpack', 'jetpack' ), 'manage_options', $jetpack_slug, null, $jetpack_icon, $position );
+
+		// Maintain id for jQuery selector.
+		$menu[ $position ][5] = 'toplevel_page_jetpack'; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
+		remove_menu_page( 'jetpack' );
+		remove_submenu_page( 'jetpack', esc_url( Redirect::get_url( 'calypso-backups' ) ) );
+
+		$this->migrate_submenus( 'jetpack', $jetpack_slug );
+
+		add_submenu_page( $jetpack_slug, esc_attr__( 'Activity Log', 'jetpack' ), __( 'Activity Log', 'jetpack' ), 'manage_options', 'https://wordpress.com/activity-log/' . $domain, null, 5 );
+		add_submenu_page( $jetpack_slug, esc_attr__( 'Backup', 'jetpack' ), __( 'Backup', 'jetpack' ), 'manage_options', 'https://wordpress.com/backup/' . $domain, null, 10 );
+		add_submenu_page( $jetpack_slug, esc_attr__( 'Scan', 'jetpack' ), __( 'Scan', 'jetpack' ), 'manage_options', 'https://wordpress.com/scan/' . $domain, null, 15 );
```

Comment:
In the current implementation, those pages are not always available; they only appear once you have a matching plan. Maybe we would want to do the same thing here?

Reply:
Yeah we should probably handle those two entirely in `Automattic\Jetpack\Scan\Admin_Sidebar_Link`. I'll remove them from here

Reply:
1fa29ed